SeriesCruncher/app/src/main/java/com/balazs_csernai/seriescruncher/seriesdetails/ui/SeriesDetailsScreenImpl.java

```diff
@@ -21,38 +23,55 @@
 /**
  * Created by Erik_Markus_Kramli on 2016-01-13.
  */
-public class SeriesDetailsScreenImpl implements SeriesDetailsScreen {
+public class SeriesDetailsScreenImpl implements SeriesDetailsScreen, SmartAppBarLayout.AppBarChangeListener {
 
-    @InjectView(R.id.poster_img)
-    ImageView poster;
+    @InjectView(R.id.appbar)
+    SmartAppBarLayout appbar;
+
+    @InjectView(R.id.collapsing_toolbar)
+    CollapsingToolbarLayout collapsingToolbar;
+
+    @InjectView(R.id.toolbar)
+    Toolbar toolbar;
 
     @InjectView(R.id.title)
     TextView title;
 
+    @InjectView(R.id.poster_img)
+    ImageView poster;
+
     @InjectView(R.id.episodes_recyclerview)
     RecyclerView episodesRecyclerView;
 
-    private final Activity activity;
+    private final AppCompatActivity activity;
     private final EpisodeAdapter adapter;
+    private final SmartLayoutManager layoutManager;
 
     @Inject
-    public SeriesDetailsScreenImpl(Activity activity, Provider<EpisodeAdapter> adapterProvider) {
+    public SeriesDetailsScreenImpl(AppCompatActivity activity, Provider<EpisodeAdapter> adapterProvider) {
         this.activity = activity;
         this.adapter = adapterProvider.get();
+        layoutManager = new SmartLayoutManager(activity);
```

Comment:
We provide constructor dependencies via injection. Please use that approach for layout manager also.